packages/anoncreds-rs/src/services/AnonCredsRsHolderService.ts

```diff
@@ -212,8 +214,14 @@ export class AnonCredsRsHolderService implements AnonCredsHolderService {
         )
       }
 
+      const isLegacyIdentifier = credentialOffer.cred_def_id.match(legacyIndyCredentialDefinitionIdRegex)
+      if (!isLegacyIdentifier && useLegacyProverDid) {
+        throw new AriesFrameworkError('Cannot use legacy prover_did with non-legacy identifiers')
+      }
+
       createReturnObj = CredentialRequest.create({
-        entropy: anoncreds.generateNonce(), // FIXME: find a better source of entropy
+        entropy: !useLegacyProverDid || !isLegacyIdentifier ? anoncreds.generateNonce() : undefined, // FIXME: find a better source of entropy
```

Comment:
Another option for generating entropy can be calling `agentContext.wallet.generateWalletKey`, which in AskarWallet creates an ephemeral key and converts its private key to a base58 string (I guess IndySdkWallet does more or less the same thing). However I'm not sure if it's good to rely on wallet implementation for that (unless we strictly specify the requirements for such `generateWalletKey` in API).

Reply:
It feels like we're misusing the generateWalletKey in that case. I think a nonce should provide enough randomness right?

Reply:
IMHO the nonce generated by anoncreds, apart from being convenient for us, is enough for the purposes of giving some entropy in credential generation. But just trying to find an alternative that provides a true 'alphanumeric' random string as stated in the spec.